zzk/service/hostregistry.go

```diff
@@ -77,11 +79,20 @@ type HostRegistryListener struct {
 
 // InitHostRegistry initializes the host registry
 func InitHostRegistry(conn client.Connection) error {
-	err := conn.CreateDir(hostregpath())
-	if err == client.ErrNodeExists {
-		return nil
+	done := make(chan error, 1)
+	path := hostregpath()
+	go func() {
+		done <- conn.CreateDir(path)
+	}()
+	select {
+	case err := <-done:
+		if err == client.ErrNodeExists {
+			return nil
+		}
+		return err
+	case <-time.After(5 * time.Second):
+		return fmt.Errorf("Could not create %s within 5 seconds", path)
```

Comment:
What happens to zookeeper and go routines if you loop forever?  Is there a scenario where CreateDir is not returning?